### PR TITLE
Remove explainer configs and organize manifests

### LINKED ISSUE #884: Add kustomize patch to remove `Namespace` created by kserve
**Describe the bug**
kserve[ creates namespace](https://github.com/opendatahub-io/odh-manifests/blob/master/kserve/kserve-built/kserve-built.yaml#L2) object which is not required for ODH deployment. Add patch that removes the namespace creation [similar to other components](https://github.com/opendatahub-io/odh-manifests/blob/master/odh-notebook-controller/kf-notebook-controller/overlays/openshift/remove_namespace_patch.yaml).

### LINKED ISSUE #885: Exclude KServe Alibi and art-explainer images and usages
**Is your feature request related to a problem? Please describe.**
We don't need alibi and art-explainer in Kserve for now so we should exclude the usage from the manifests (and stop producing the images)

**Describe the solution you'd like**
It should be possible to apply KServe manifests without having alibi and art-explainer images

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by israel-hdez**
## Description

* References to Alibi and ART explainers are removed from the `inferenceservice-config` ConfigMap.
* User cluster roles are moved into an overlay in opendatahub-io/kserve: opendatahub-io/kserve#35
* Namespace resource is excluded

Fixes #884 
Fixes #885 
Related to #887

## How Has This Been Tested?

* Run `kustomize build ./kserve/base > before.yaml` over the current `master` branch.
* Run `kustomize build ./kserve/base > after.yaml` over this PR.
* Running `diff -u before.yaml after.yaml` generates the following diff:
```diff
--- before.yaml 2023-07-21 13:13:48.120286546 -0600
+++ after.yaml  2023-07-21 13:23:33.518921123 -0600
@@ -1,13 +1,3 @@
-apiVersion: v1
-kind: Namespace
-metadata:
-  labels:
-    app.kubernetes.io/part-of: kserve
-    control-plane: kserve-controller-manager
-    controller-tools.k8s.io: "1.0"
-    istio-injection: disabled
-  name: opendatahub
----
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
@@ -18372,17 +18362,7 @@
     {
       "defaultDeploymentMode": "Serverless"
     }
-  explainers: |-
-    {
-        "alibi": {
-            "image" : "quay.io/opendatahub/kserve-alibi-explainer",
-            "defaultImageVersion": "v0.10.2"
-        },
-        "art": {
-            "image" : "quay.io/opendatahub/kserve-art-explainer",
-            "defaultImageVersion": "latest"
-        }
-    }
+  explainers: '{}'
   ingress: |-
     {
         "ingressGateway" : "knative-serving/knative-ingress-gateway",
@@ -18437,10 +18417,6 @@
 apiVersion: v1
 data:
   kserve-agent: quay.io/opendatahub/kserve-agent:v0.10.2
-  kserve-alibi-explainer: quay.io/opendatahub/kserve-alibi-explainer
-  kserve-alibi-explainer-version: v0.10.2
-  kserve-art-explainer: quay.io/opendatahub/kserve-art-explainer
-  kserve-art-explainer-version: latest
   kserve-controller: quay.io/opendatahub/kserve-controller:v0.10.2
   kserve-router: quay.io/opendatahub/kserve-router:v0.10.2
   kserve-storage-initializer: quay.io/opendatahub/kserve-storage-initializer:v0.10.2
```
* Visually inspect that the differential indicates that the `Namespace` resource is removed.
* Visually inspect that the differential suggests that the `inferenceservice-config` has empty explainers configuration.
* Visually inspect that the differential suggests that the `kserve-parameters` ConfigMap no longer includes explainer parameters.
* Extract the `inferenceservice-config` ConfigMap from `after.yaml`, apply it to an existent KServe installation, and restart and confirm that the KServe controller runs normally.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: kserve/base/inferenceservice-config-patch.yaml [hunk 1]
```diff
@@ -4,17 +4,7 @@ metadata:
...
-        "alibi": {
-            "image" : "$(kserve-alibi-explainer)",
-            "defaultImageVersion": "$(kserve-alibi-explainer-version)"
-        },
-        "art": {
-            "image" : "$(kserve-art-explainer)",
-            "defaultImageVersion": "$(kserve-art-explainer-version)"
-        }
-    }
+  explainers: "{}"
```
Comment (Jooho) [inline kserve/base/inferenceservice-config-patch.yaml:7]: `kustomize build` shows 
~~~
 deploy: |-
    {       
      "defaultDeploymentMode": "Serverless"
    }       
  explainers: '{}'
  ingress: |-   
    {             
        "ingress.....
~~~
I didn't test it but I am not sure if `{}` is ok to deploy kserve and a model? If it works, I am ok with it for now before we integrate with TrustyAI.

Comment (israel-hdez) [reply kserve/base/inferenceservice-config-patch.yaml:7]: We should be safe if we always omit using explainers.

As soon as we try to use some explainer, we may have issues. But we are not supporting them at the moment.

**Review by Jooho — APPROVED** [review]
/lgtm

**Pull request merged**